src/forces.f90

```diff
@@ -13,522 +13,832 @@
+        if (istret.eq.0) then 
+          jcvlw(iv) = nint(yld(iv)/dy)+1
+          jcvup(iv) = nint(yud(iv)/dy)+1
+        else
+          stp1=0
+          stp2=0
+          do h = 1, ny-1  
+            if ((-yp(h+1)-yp(h)+two*yld(iv)).lt.(yld(iv)-yp(h)).and.(stp1.eq.0)) then 
+              jcvlw(iv) = h+1
+              stp1=1
+            endif
+            if ((-yp(h+1)-yp(h)+two*yud(iv)).lt.(yud(iv)-yp(h)).and.(stp2.eq.0)) then
+              jcvup(iv) = h
+              stp2=1 
+            endif
+          enddo
+        endif
+        icvlf_lx(iv) = icvlf(iv)
+        icvrt_lx(iv) = icvrt(iv)
+        jcvlw_lx(iv) = max(jcvlw(iv)+1-xstart(2),1)
+        jcvup_lx(iv) = min(jcvup(iv)+1-xstart(2),xsize(2))
+        kcvfr_lx(iv) = max(kcvfr(iv)+1-xstart(3),1)
+        kcvbk_lx(iv) = min(kcvbk(iv)+1-xstart(3),xsize(3))
+        icvlf_ly(iv) = max(icvlf(iv)+1-ystart(1),1)
+        icvrt_ly(iv) = min(icvrt(iv)+1-ystart(1),ysize(1))
+        jcvlw_ly(iv) = jcvlw(iv)
+        jcvup_ly(iv) = jcvup(iv)
+        kcvfr_ly(iv) = max(kcvfr(iv)+1-ystart(3),1)
+        kcvbk_ly(iv) = min(kcvbk(iv)+1-ystart(3),ysize(3))
+     enddo
+     
+     if (nrank==0) then
+        write(*,*) '========================Forces============================='
+        write(*,*) '     (icvlf)      (icvrt)    (kcvbk)    (kcvfr)'
+        write(*,*) '  (jcvup) B____________C     B`_____________B'
+        write(*,*) '          \            \  |  \              \'
+        write(*,*) '          \     __     \  |  \     ____     \'
+        write(*,*) '          \    \__\    \  |  \     \___\    \'
+        write(*,*) '          \            \  |  \              \'
+        write(*,*) '          \       CV   \  |  \    (Front)   \'
+        write(*,*) '  (jcvlw) A____________D  |  A`_____________A'
+        do iv=1,nvol
+           write(*,"(' Control Volume     : #',I1)") iv
+           write(*,"('     xld, icvlf     : (',F6.2,',',I6,')')") xld(iv), icvlf(iv)
+           write(*,"('     xrd, icvrt     : (',F6.2,',',I6,')')") xrd(iv), icvrt(iv)
+           write(*,"('     yld, jcvlw     : (',F6.2,',',I6,')')") yld(iv), jcvlw(iv)
+           write(*,"('     yud, jcvup     : (',F6.2,',',I6,')')") yud(iv), jcvup(iv)
+           write(*,"('     zfr, kcvfr     : (',F6.2,',',I6,')')") zfr(iv), kcvfr(iv)
+           write(*,"('     zbk, kcvbk     : (',F6.2,',',I6,')')") zbk(iv), kcvbk(iv)
+        enddo
+        write(*,*) '==========================================================='
+     endif
+ 
+     call decomp_2d_init_io(io_restart_forces)
+     call decomp_2d_register_variable(io_restart_forces, "ux01", 1, 0, 0, mytype)
+     call decomp_2d_register_variable(io_restart_forces, "uy01", 1, 0, 0, mytype)
+     call decomp_2d_register_variable(io_restart_forces, "ux11", 1, 0, 0, mytype)
+     call decomp_2d_register_variable(io_restart_forces, "uy11", 1, 0, 0, mytype)
+     
+   end subroutine init_forces
+ 
+   subroutine restart_forces(itest1)
+ 
+     USE decomp_2d
```

Comment:
This is not needed

Reply:
Same, this is also present in the latest version of the code.